src/gpu/intel/ocl/reusable_softmax.cl

```diff
@@ -70,3 +71,157 @@ reusable_softmax_fwd_generic(__global SRC_DATA_T *src, __global DST_DATA_T *dst,
         dst[c - begin] = TO_DST(unscaled * scale);
     }
 }
+#endif
+
+#define DIVUP(a, b) (((a) + (b)-1) / (b))
+
+#define STORE_FLOAT8(prefix, ptr, val) \
+    WRITE_BLOCK8(prefix, (__global BLOCK_T(ALIAS(prefix)) *)(ptr), \
+            DATA_TO_BLOCK8(prefix, FLOAT_TO_DATA8(prefix, val)))
+
+#define STORE_DOUBLE8(prefix, ptr, val) \
+    WRITE_BLOCK8(prefix, (__global BLOCK_T(ALIAS(prefix)) *)(ptr), \
+            DATA_TO_BLOCK8(prefix, DOUBLE_TO_DATA8(prefix, val)))
+
+#if DST_DT_F64
+#define UP_CASE_DATA DOUBLE
+#define COMMON_DATA_T double
+#define COMMON_DATA_MAX DBL_MAX
+#define COMMON_DATA_ZERO 0.0
+#else
+#define UP_CASE_DATA FLOAT
+#define COMMON_DATA_T float
+#define COMMON_DATA_MAX FLT_MAX
+#define COMMON_DATA_ZERO 0.0f
+#endif
+
+#define COMMON_DATA_TO_X(x, y) CONCAT2(DATA_TO_, UP_CASE_DATA)(x, y)
+#define COMMON_STORE_DATA8(x, y, z) CONCAT3(STORE_, UP_CASE_DATA, 8)(x, y, z)
```

Comment:
Could we extended the `ocl_io.h` load/store functionality and switch these kernels to use `ocl_io.h`?

Reply:
I added an experimental interface in the branch `rjoursle/block_load` that should be usable for this.